sdk/core/System.ClientModel/README.md

```diff
@@ -25,67 +24,144 @@ None needed for `System.ClientModel`.
 
 ### Authenticate the client
 
-The `System.ClientModel` package provides a `KeyCredential` type for authentication.
+The `System.ClientModel` package provides an `ApiKeyCredential` type for authentication.
 
 ## Key concepts
 
-The main shared concepts of `System.ClientModel` include:
+The main concepts in `System.ClientModel` include:
 
 - Configuring service clients (`ClientPipelineOptions`).
```

Comment:
I think it would be good if this section started with key concepts of this library and then separately key concepts of clients using this library. I think readers might be confused why this section talks about clients and there are no clients in this library. BTW, the key concepts in this library that would be good to have sections for are: the pipeline, model persistence APIs, types that show in public surface of clients (ClientResponse/RequestOptions/BiodyContent).

Reply:
I restructured the intro based on your suggested sections and added samples for each of the sections listed.  Let me know if you have further suggestions based on those updates!